Wikipedia/assets/codemirror/codemirror-editTextSelection.js

```diff
@@ -6,92 +6,59 @@ class SelectedAndAdjacentText {
-    const textAfterSelectedTextPattern = replaceSpaceWith(wordsAfter, patternForSpace)
-
-    // Attempt to locate wikitext selection based on the non-wikitext context strings above.
-    const atLeastOneNonWordOrOptionalStringPattern = '(?:\\W+|.*)'
-    const pattern = `(${textBeforeSelectedTextPattern.length > 0 ? '.*?' : ''}${textBeforeSelectedTextPattern}${atLeastOneNonWordOrOptionalStringPattern})(${selectedTextPattern})${atLeastOneNonWordOrOptionalStringPattern}${textAfterSelectedTextPattern}`
-    const regex = new RegExp(pattern, 's')
-
-    return regex
-  }
+const wordsOnly = string => string.replace(/\(.*?\)/g, '').replace(/{{.*}}/g, '').replace(/\W+/g, ' ').trim()
+const replaceSpaceWith = string => string.replace(/\s+/g, '\\W*(?:(?:[<{][^>]*[}>])*)\\W*')
+
+const adjacentComparisonWordCount = 6
+const adjacentCharsToGather = 200
+
+const scoreForMatch = (match, lastIndex, wordsBeforeFromHTML, wordsAfterFromHTML) => {
+  const wordsBeforeFromWikitext = getNumberOfWordsFromEndOfString(wordsOnly(match.input.substring(Math.max(0, match.index - adjacentCharsToGather), match.index)), adjacentComparisonWordCount)
+  const wordsAfterFromWikitext = getNumberOfWordsFromBeginningOfString(wordsOnly(match.input.substring(lastIndex, lastIndex + adjacentCharsToGather)), adjacentComparisonWordCount)
+  const wordsAfterScore = calculateScore(wordsAfterFromHTML, wordsAfterFromWikitext)
+  const wordsBeforeScore = calculateScore(wordsBeforeFromHTML.slice().reverse(), wordsBeforeFromWikitext.slice().reverse())
+  return wordsBeforeScore + wordsAfterScore
 }
 
 const wikitextRangeForSelectedAndAdjacentText = (selectedAndAdjacentText, wikitext) => {
-  const regex = selectedAndAdjacentText.regexForLocatingSelectedTextInWikitext(wikitext)
-  if (regex === null) {
-    return null
+  const wordsBeforeFromHTML = getNumberOfWordsFromEndOfString(selectedAndAdjacentText.textBeforeSelectedText, adjacentComparisonWordCount)
+  const wordsAfterFromHTML = getNumberOfWordsFromBeginningOfString(selectedAndAdjacentText.textAfterSelectedText, adjacentComparisonWordCount)
+  const selectedTextRegexPattern = `(${selectedAndAdjacentText.selectedText.replace(/\s+/g, '\\W*(?:(?:[<{][^>]*[}>])*)\\W*')})`
+  const selectedTextRegex = new RegExp(selectedTextRegexPattern, 'gs')
+
+  let bestScoredMatch = null
+  while ((match = selectedTextRegex.exec(wikitext)) !== null) {
+    const thisMatchScore = scoreForMatch(match, selectedTextRegex.lastIndex, wordsBeforeFromHTML, wordsAfterFromHTML)
+    if (bestScoredMatch === null || thisMatchScore > bestScoredMatch.score) {
+      bestScoredMatch = {
+        match, 
+        score: thisMatchScore,
+        matchedWikitextBeforeSelection: {start: 0, end: match.index},
+        matchedWikitextSelection: {start: match.index, end: selectedTextRegex.lastIndex}
+      }    
+    }
   }
-  const match = wikitext.match(regex)
-  if (match === null) {
+
+  if (bestScoredMatch === undefined) {
```

Comment:
`null === undefined` is `false` so this fails when `bestScoredMatch` remains `null`

Reply:
doh! 😜

Reply:
fixed!